### PR TITLE
Remove log4j usage

### PR DESCRIPTION
**Pull request by cmahnke**
See https://github.com/onthegomap/planetiler/discussions/773#discussioncomment-8037396

### CODE REVIEW

**Comment by cmahnke** [comment]
Please don't merge yet.

**Comment by cmahnke** [comment]
I've checked locally and downloading there works, can you please rerun the failing checks?

File: planetiler-core/src/main/java/com/onthegomap/planetiler/expression/Expression.java [hunk 1]
```diff
@@ -362,7 +362,7 @@ static MatchAny from(String field, BiFunction<WithTags, String, Object> valueGet
 
       return new MatchAny(field, values,
         Set.copyOf(exactMatches),
-        patterns.isEmpty() ? null : Pattern.compile("(" + Strings.join(patterns, '|') + ")"),
+        patterns.isEmpty() ? null : Pattern.compile("(" + Joiner.on('|').join(patterns) + ")"),
```
Comment (bbilger) [inline planetiler-core/src/main/java/com/onthegomap/planetiler/expression/Expression.java:365]: not sure if I miss anything but there's IMO no need for guava
would personally either do
`Pattern.compile("(" + String.join("|", patterns) + ")")`
or even "better"
`Pattern.compile(patterns.stream().collect(Collectors.joining("|", "(", ")")))`

Comment (msbarry) [reply planetiler-core/src/main/java/com/onthegomap/planetiler/expression/Expression.java:365]: +1, slight preference to using built-in utilities.

Comment (cmahnke) [reply planetiler-core/src/main/java/com/onthegomap/planetiler/expression/Expression.java]: Thanks for your feedback! I've included the proposed change.

**Pull request merged**